server/src/api/read-schematic-collections.ts

```diff
@@ -22,10 +27,30 @@ interface Schematic {
   }[];
 }
 
-export function readSchematicCollections(
+export function readAllSchematicCollections(basedir: string) {
+  try {
+    const nodeModulesDir = path.join(basedir, 'node_modules');
+    const packages = listOfUnnestedNpmPackages(nodeModulesDir);
+    const schematicCollections = packages.filter(p => {
+      try {
+        return !!readJsonFile(path.join(p, 'package.json'), nodeModulesDir).json
+          .schematics;
+      } catch (e) {
+        return false;
+      }
+    });
+    return schematicCollections.map(c =>
+      readSchematicCollections(nodeModulesDir, c)
+    );
+  } catch (e) {
```

Comment:
What's the point of this try catch if we just rethrow?

Reply:
Removed it.